### PR TITLE
Initial creation of devimg

### PR DESCRIPTION
**Pull request by tgjonestx**
See comments in [devimg/makefile](./devimg/makefile) for more details on how devimg is built.

A typical invocation by zendev would look something like:
```
cdz product-asembly
cd devimg
ZENDEV_ROOT=$HOME/src/metis SRCROOT=$HOME/src/metis/src TARGET_PRODUCT=core make clean build
```

The result will be two docker images:
```
zendev/devimg:metis
zendev/devimg-base:metis
```

### CODE REVIEW

File: devimg/makefile [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+#     |
+# zenoss/zenoss-centos-base:<ver>.devtools
+#
+# zenoss-centos-base:<ver>.devtools is a variant of zenoss-centos-base that
+# contains development tools like the JDK, Maven, make and GO.
+#
+# devimg-base:<envName> is a variant of product-base with the uid/gid of the
+# zenoss user/group remapped to the uid/gid of the developer creating devimg.
+#
+# zendev/devimg is built just like the standard product images, except:
```
Comment (jplouis) [inline devimg/makefile]: Products is also linked, or should be.

File: devimg/makefile [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ZENPACK_SRC ?= $(SRCROOT)
+PRODBIN_SRC ?= $(SRCROOT)/zenoss-prodbin
+
+#
+# four major steps to creating the devimg:
+# 1. build-devbase - Create a devimg-base image starting the same as product-base, but
+#    built on the ".devtools" version of zenoss-centos-base.
+# 2. initialize-zenhome - Copies the contents of /opt/zenoss/* from the image
+#    into our local ZENHOME directory
+# 3. add-zenpack-file - Adds the zenpacks to the image based on a file
```
Comment (jplouis) [inline devimg/makefile]: Technically you are adding the zenpacks.json to the local zenhome that gets mounted into the container.

File: devimg/build.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# build.sh - build zendev/devimg.
+#
+# This script assumes that zendev/devimg already exists as an incomplete image;
```
Comment (jplouis) [inline devimg/build.sh]: name of image is wrong

File: devimg/build.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+# Run create_zenoss, record the container id and CONTAINER_ID_FILE and leave
+# the image running so that we can commit the changes
+#
+docker run \
+	--cidfile ${CONTAINER_ID_FILE} \
+	-v ${ZENDEV_ROOT}/zenhome:/opt/zenoss \
+	-v ${ZENDEV_ROOT}/var_zenoss:/var/zenoss \
+	-v ${SRCROOT}:/mnt/src \
+	-v ${PRODBIN_SRC}/Products:/opt/zenoss/Products \
```
Comment (jplouis) [inline devimg/build.sh]: Why isn't all of zenhome mounted in?

File: devimg/build.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	-v ${PRODBIN_SRC}/devimg:/opt/zenoss/devimg \
+	-v ${ZENPACK_SRC}:/opt/zenoss/packs \
```
Comment (jplouis) [inline devimg/build.sh]: Why don't we make these softlinks inside of the mounted in zenhome so you just bind mount zenhome?  That would be more similar to the current devimg.

File: devimg/makefile [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+	@echo "Building $(BASE_TAG) ..."
+	BASE_TAG=$(BASE_TAG) ./build-devbase.sh
+
+# Initialize our local ZENHOME using the contents from the initial devimg
+initialize-zenhome: make-dirs
+	@echo "Initializing ZENHOME from container contents..."
+	docker run --rm \
+		-v $(ZENDEV_ROOT)/zenhome:/mnt/local-zenhome \
+		-t $(BASE_TAG) \
+		bash -c "rm -rf /mnt/local-zenhome/*; cp -rp $(CONTAINER_ZENHOME)/* /mnt/local-zenhome"
```
Comment (jplouis) [inline devimg/makefile]: You are copying out zenhome/Products. It should probably be removed and be a soft link to zenoss-prodbin/Products

File: devimg/build.sh (hunk 5 above)
Comment (tgjonestx) [inline devimg/build.sh]: Good point - I was so heads down on changes for incremental testing, I guess I lost sight of the trees.

**Pull request merged**